cep-XXXX.md

```diff
@@ -0,0 +1,216 @@
+
+```js
+{
+    "build": str, // build string
+    "build_number": int,
+    "channel": str, // URL to source channel, no subdir
+    "constrains": list[str], // str must be a "conda-build form" match spec
+    "depends": list[str], // str must be a "conda-build form" match spec
+    "extracted_package_dir": str, // absolute path to extracted contents
+    "files": list[str], // list of paths in artifact, relative to $CONDA_PREFIX, forward-slash normalized
+    "fn": str, // {name}-{version}-{build}.{extension}
+    "license": str, // SPDX expression preferred
+    "link": { // How the package was linked into the prefix
+        "source": str, // original path
+        "type": (1|2|3|4), // hardlink, softlink, copy, directory
+    },
+    "md5": str, // 32-char hex string
+    "name": str, // package name
+    "noarch": "python", // optional, values: python, generic
+    "package_tarball_full_path": str, // absolute path to artifact
+    "package_type": "noarch_python", // optional, values: noarch_python
+    "paths_data": { // Artifact contents + generated files in $CONDA_PREFIX
+        "paths": [  // one dict per file:
+            {
+                "_path": str, // relative path of file to $CONDA_PREFIX, forward-slash normalized
+                "file_mode": str, // optional; one of 'text' or 'binary'
+                "no_link": bool, // optional, whether to force copy or allow link
+                "path_type": str, // one of: softlink, hardlink, directory; MAY also be one of generated types: pyc_file, unix_python_entry_point, windows_python_entry_point_script, windows_python_entry_point_exe, linked_package_record
+                "prefix_placeholder": str, // optional; to be replaced with $CONDA_PREFIX
+                "sha256": str, // optional if generated, 64-char hex string of file in cache
+                "sha256_in_prefix": str, // optional if generated, 64-char hex string of file in prefix (after placeholder has been replaced)
+                "size_in_bytes": int // optional, if generated
+            },
+            ...
+            ]
+        "paths_version": 1
+    },
+    "requested_spec": str, // MatchSpec asked by user; use "None" if not requested (transitive)
+    "sha256": str, // 64-char hex string
+    "size": int, // in bytes
+    "subdir": str, // one of KNOWN_SUBDIRS
+    "timestamp": int, // in ms
+    "url": str, // download URL of the package
+    "version": str, // package version
+}
+```
+
+The serialized `PrefixRecord` information SHOULD match the relevant fields in the most up-to-date repodata information available for the package (`depends` and `constrains` are of particular importance due to repodata patching). This is generally the channel's `repodata.json`, but it MAY also be an alternative source like the serialized metadata in a lockfile. The package's `info/index.json` MAY be used as a fallback if no other sources are available.
```

Comment:
> but it MAY also be an alternative source like the serialized metadata in a lockfile

I would like to see this change to a MUST requirement when the metadata is provided in a lockfile. IMHO providing static reliable metadata is one of the key reasons for a lockfile in conda.

Reply:
I think the MUST/SHOULD distinction should come earlier in the paragraph, as in:
```suggestion
The serialized `PrefixRecord` information MUST match the relevant fields in the most up-to-date repodata information available for the package (`depends` and `constrains` are of particular importance due to repodata patching). This is generally the channel's `repodata.json`, but it MAY also be an alternative source like the serialized metadata in a lockfile. The package's `info/index.json` MAY be used as a fallback if no other sources are available.
```

Otherwise we need to rephrase this and establish a set of rules to follow.

Reply:
Worked on this part a little bit. Let me know if it's now better!

Reply:
Thanks. This section looks greats now.